src/components/Notifications/EventLog/EventLogTreeFilter.tsx

```diff
@@ -0,0 +1,214 @@
+        };
+    });
+
+    return init;
+};
+
+export const EventLogTreeFilter: React.FunctionComponent<EventLogTreeFilterProps> = (props) => {
+    const { groups, placeholder, filters, updateFilters } = props;
+
+    const initialize = React.useMemo(() => initTreeNodeById(groups, filters), [ groups, filters ]);
+
+    const [ treeNodeById, setTreeNodeById ] = React.useState<TreeNodeDict>(initialize);
+    const [ isToggled, setIsToggled ] = React.useState(false);
+
+    const treeDataArray = React.useMemo(() => !!treeNodeById ? Object.values(treeNodeById) : [], [ treeNodeById ]);
+
+    React.useEffect(() => {
+        if (groups.length !== 0) {
+            setTreeNodeById(produce((prev) => {
+                // In case Bundles request is pending during Tree View render
+                if (Object.keys(prev).length === 0) {
+                    return initialize;
+                }
+                else if (filters.length === 0) {
+                    return initialize;
+                }
+                else {
+                    filters.forEach(activeFilter => {
+                        const treeNode = prev[activeFilter.bundleId];
+                        const activeChips = activeFilter.chips.map(chip => chip.value);
+
+                        treeNode.children?.forEach(childNode => {
+                            childNode.checkProps.checked = activeChips.includes(childNode.id);
+                        });
+
+                        if (allChildrenChecked(treeNode)) {
+                            treeNode.checkProps.checked = true;
+                        }
+                        else if (childChecked(treeNode)) {
+                            treeNode.checkProps.checked = null;
+                        }
+                        else {
+                            treeNode.checkProps.checked = false;
+                        }
+                    });
+                }
+            }));
+        }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, [ groups, filters ]);
+
+    const flattenTree = React.useMemo(() => {
+        const flatTreeDataArray = produce(treeDataArray, (prev) => {
+            prev.forEach(treeNode => {
+                if (!!treeNode.children) {
+                    prev.push(...treeNode.children);
+                }
+            });
+        });
+
+        return flatTreeDataArray;
+    }, [ treeDataArray ]);
+
+    React.useEffect(() => {
+        const groupNames = groups.map(group => group.name);
+        const activeParentFilters = flattenTree.filter(treeNode => {
+            const isActive = treeNode.checkProps.checked;
+            return (!!isActive || isActive === null) && groupNames.includes(treeNode.id);
+        });
+
+        updateFilters(produce((prev) => {
+            if (prev.length === 0 && activeParentFilters.length === 0) {
+                return prev;
+            }
+
+            else if (prev.length !== 0 && activeParentFilters.length === 0) {
```

Comment:
```suggestion
            } else if (prev.length !== 0 && activeParentFilters.length === 0) {
```
or at least lest keep them closer

Reply:
I will change to aforementioned style as stated in my reply above.